agent/consul/catalog_endpoint_test.go

```diff
@@ -2765,6 +2765,109 @@ node_prefix "" {
 	return
 }
 
+// TestCatalog_Register_DenyPeeringRegistration makes sure that users cannot send structs.RegisterRequest
+// with a PeerName in any part of the request.
+func TestCatalog_Register_DenyPeeringRegistration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	_, s := testServerWithConfig(t)
+	codec := rpcClient(t, s)
+
+	// we will add PeerName to copies of arg
+	arg := structs.RegisterRequest{
+		Datacenter: "dc1",
+		Node:       "foo",
+		Address:    "127.0.0.1",
+		Service: &structs.NodeService{
+			Service: "db",
+			Tags:    []string{"primary"},
+			Port:    8000,
+		},
+		Check: &structs.HealthCheck{
+			CheckID:   types.CheckID("db-check"),
+			ServiceID: "db",
+		},
+		Checks: structs.HealthChecks{
+			&structs.HealthCheck{
+				CheckID:   types.CheckID("db-check"),
+				ServiceID: "db",
+			},
+		},
+	}
+
+	type testcase struct {
+		name      string
+		reqCopyFn func(arg *structs.RegisterRequest) structs.RegisterRequest
+	}
+
+	/*
+		NOTE that there is enforcement for one and only PeerName to be present in the state store.
```

Comment:
```
NOTE that there is enforcement for one and only PeerName to be present in the state store.
```
Could you clarify what this means? Unsure what the implications are

Reply:
I _think_ this is mostly applicable to the "peer names everywhere" test case that sets multiple different peer names at each point in the register request. That register request would be deemed invalid and any raftApply for it would fail.

Reply:
So yeah matt's spot on above; it refers to the validations we do at the store level ... but that knowledge comes form an open box thinking; I removed the comment since it's not that applicable to what we are testing. We want to block any mention of `PeerName`, regardless of the values of the field.